### ISSUE TITLE
Add missing Dockerfiles for the data assets of the METplus 3.1 release.

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/METplus#589]
## Describe the Problem ##

The METplus 3.1 release contains Dockerfile to construct data volumes for 3 of the release data assets in:
METplus/internal_tests/docker_data

This task is to:

1. Create 5 new Dockerfiles for the other 5 release data assets listed at:
https://github.com/dtcenter/METplus/releases/tag/v3.1
2. Run the Dockerfiles to construct the data volumes.
3. Construct a 9th data volume that is the union of the other 8.
3. Push the data volumes to the metplus_data repository of DockerHub and tag them as version 3.1.
https://hub.docker.com/repository/docker/dtcenter/metplus_data

This data is needed for the METplus-Training Docker setup described here:
https://dtcenter.github.io/METplus-Training/modules/Environment/docker.html

The instructions need to be updated to tell users how to pull/use these data volumes.

Make this change in both the master_v3.1 and develop branches.

### Expected Behavior ###
*Provide a clear and concise description of what you expected to happen here.*

### Environment ###
Describe your runtime environment:
*1. Machine: (e.g. HPC name, Linux Workstation, Mac Laptop)*
*2. OS: (e.g. RedHat Linux, MacOS)*
*3. Software version number(s)*

### To Reproduce ###
Describe the steps to reproduce the behavior:
*1. Go to '...'*
*2. Click on '....'*
*3. Scroll down to '....'*
*4. See error*
*Post relevant sample data following these instructions:*
*https://dtcenter.org/community-code/model-evaluation-tools-met/met-help-desk#ftp*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required - John HG
- [x] Select **scientist(s)** or **no scientist** required - None required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones
- [x] Select **milestone**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
Need to update the METplus-Training environment docker directions.

## Bugfix Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above.
- [x] Fork this repository or create a branch of **master_\<Version>**.
Branch name:  `bugfix_<Issue Number>_master_<Version>_<Description>`
- [x] Fix the bug and test your changes.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **master_\<Version>** and link the pull request to this issue.
Pull request: `bugfix <Issue Number> master_<Version> <Description>`
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Complete the steps above to fix the bug on the **develop** branch and link the pull request to this issue.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
- [x] Close this issue.

**Comment by JohnHalleyGotway** [comment]
Created a DockerHub metplus-data repository:
https://hub.docker.com/repository/docker/dtcenter/metplus-data
This makes the 'dash' naming convention consistent with metplus-training and metexpress-...
Recommend removing existing metplus_data repository.
Replaced individual Dockerfiles for each tarball with a single parameterized Dockerfile and shell script to call it multiple times. Does not actually push images to DockerHub though. That's still manual.

**Comment by georgemccabe** [comment]
metplus_data on DockerHub has been removed.

**Comment by JohnHalleyGotway** [comment]
Rather than having many nearly identical Dockerfiles (one for each sample data tarball), I parameterized a single Dockerfile to handle this.
VERSION = METplus version number for which this data is posted
TARFILE = Sample data tarfile name (e.g. https://github.com/dtcenter/METplus/releases/download/${VERSION}/${TARFILE})
MOUNTPT = Path to the directory for which the data volume should be created

The metplus_sample_data ascii file contains one line for each tarball and named as:
{name}:{version}:{tarfile}:{path to mount directory}

The build_docker_images.sh script loops over the metplus_sample_data entries and creates a Docker image for each.

I did NOT include logic in the shell script for pushing these images to DockerHub. So that's still a manual process. But that could be easily added.

I also did NOT create a single image that contains the union of all of the data. That image would just be way too big. Instead, I tested to make sure that the data volumes work together nicely using multiple "--volumes-from" options... and they do.
